From: John W. Linville <linville@redhat.com>
Date: Wed, 3 Sep 2008 14:19:14 -0400
Subject: [wireless] compiler warning fixes for mac80211 update
Message-id: 20080903181913.GE5874@redhat.com
O-Subject: Re: [RHEL5 patch 1/6] infrastructure changes to support mac80211 update
Bugzilla: 438391
RH-Acked-by: Jon Masters <jcm@redhat.com>

> >  #define inline		inline		__attribute__((always_inline))
> >  #define __inline__	__inline__	__attribute__((always_inline))
> >  #define __inline	__inline	__attribute__((always_inline))
> >  #define __deprecated			__attribute__((deprecated))
> > +#define __packed                       __attribute__((packed))
> >  #define  noinline			__attribute__((noinline))
> >  #define __attribute_pure__		__attribute__((pure))
> >  #define __attribute_const__		__attribute__((__const__))
>
> This chunk is generating a lof of 'redifined' warning messages.  Perhaps
> we should drop it?  Despite the fact I have already included the patch
> into -107.el5.

Dropping it will break the mac80211 stuff.  I looked into and I don't
see any other definitions of __aligned.  It does look like hfs and
hfsplus filesystems have definitions of __packed, so I propose the
patch below.

Thanks!

John

diff --git a/fs/hfs/hfs.h b/fs/hfs/hfs.h
index c984cf7..6f194d0 100644
--- a/fs/hfs/hfs.h
+++ b/fs/hfs/hfs.h
@@ -83,8 +83,6 @@
 
 /*======== HFS structures as they appear on the disk ========*/
 
-#define __packed __attribute__ ((packed))
-
 /* Pascal-style string of up to 31 characters */
 struct hfs_name {
 	u8 len;
diff --git a/fs/hfsplus/hfsplus_raw.h b/fs/hfsplus/hfsplus_raw.h
index 4920553..fe99fe8 100644
--- a/fs/hfsplus/hfsplus_raw.h
+++ b/fs/hfsplus/hfsplus_raw.h
@@ -15,8 +15,6 @@
 
 #include <linux/types.h>
 
-#define __packed __attribute__ ((packed))
-
 /* Some constants */
 #define HFSPLUS_SECTOR_SIZE        512
 #define HFSPLUS_SECTOR_SHIFT         9
